jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-kops-aws.yaml

```diff
@@ -0,0 +1,67 @@
+- job-template:
+    name: 'kubernetes-e2e-kops-{suffix}'
+    node: '{jenkins_node}'
+    description: '{description} Test owner: {test-owner}.'
+    jenkins_node: 'e2e'
+    disabled: '{obj:disable_job}'
+    properties:
+        - build-discarder:
+            days-to-keep: 7
+    triggers:
+        - reverse:
+            jobs: '{trigger-job}'
+            result: success
+        - timed: '{cron-string}'
+    publishers:
+        - junit-publisher
+        - email-ext:
+            recipients: '{emails}'
+        - gcs-uploader
+    provider-env: |
+        export E2E_OPT="--deployment kops --kops $WORKSPACE/kops -kops-cluster {suffix}.test-aws.k8s.io -kops-state s3://k8s-kops-jenkins/"
+        export AWS_CONFIG_FILE=/workspace/.aws/credentials
+        # This is needed to be able to create PD from the e2e test
+        export AWS_SHARED_CREDENTIALS_FILE=/workspace/.aws/credentials
+        export KUBE_SSH_USER=admin
+    builders:
+        - activate-gce-service-account
+        - shell: |
+            {provider-env}
+            {job-env}
+            {post-env}
+            if ! curl -fsS --retry 3 -o "$WORKSPACE/kops.url" "https://storage.googleapis.com/kops-ci/bin/latest-ci.txt"; then
```

Comment:
The example code you give is still too complex for yaml. Conditional logic is complicated and therefore belongs in its own script, which jenkins job can call unconditionally:

`kops-runner.sh`
```
if ! kops-download.sh $WORKSPACE/kops; then
  exit 1
fi
# Whatever complex stuff you want
. $WORKSPACE/kops.env
timeout {runner}
```

`kops-aws.yaml`
```
export KUBERNETES_PROVIDER="aws"  # Pretty soon we will disallow this complexity too
bash <(curl -fsS --retry 3  "https://raw.githubusercontent.com/kubernetes/test-infra/master/jenkins/kops-runner.sh")  # Pretty soon we will automatically and always checkout test-infra at master
```

I do not want complex logic in yaml, and I'm spending a lot of effort in this space. I allowed it initially because at least in the yaml it is source controlled, but fundamentally there needs to be an easy way for someone to run your job without jenkins.

If you encapsulate this logic inside a `kops-runner.sh` or `kops-runner.go` program it will be much easier for someone to run this job the same way jenkins will: call the program.